hedera-node/docs/design/services/smart-contract-service/frictionless-airdrops-system-contracts.md

```diff
@@ -0,0 +1,166 @@
+# Frictionless Airdrops via System Contracts
+
+## Purpose
+
+[HIP-904](https://hips.hedera.com/hip/hip-904) introduced the Frictionless Airdrops feature for fungible and non-fungible tokens.
+This document will define the architecture and implementation of the `airdropToken`, `claimAirdrops`, `cancelAirdrops` and `rejectTokens` smart contract functions
+and their respective redirect function calls (`cancelAirdropFT`, `cancleAirdropNFT`, `claimAirdropFT`, `claimAirdropNFT`, `rejectTokenFT`, `rejectTokenNFTs`, `setAutomaticAssociations`) that will extend the capabilities of the Hedera Smart Contract Service (HSCS) to support frictionless airdrops.
+
+## References
+
+[HIP-904](https://hips.hedera.com/hip/hip-904) - HIP that introduces the frictionless airdrops.
+
+## Goals
+
+- Expose `airdropToken`, `claimAirdrops`, `cancelAirdrops` and `rejectTokens` as new functions in the Hedera Token Service Smart Contract.
+- Expose `cancelAirdropFT`, `cancleAirdropNFT`, `claimAirdropFT`, `claimAirdropNFT`, `rejectTokenFt`, `rejectTokenNFTs`, `setAutomaticAssociations` as new functions in a new proxy redirect token facade contract IHRC904.
+- Implement the needed HTS system contract classes to support the new functions.
+
+## Non Goals
+
+- The implementation of the HAPI operation, as it is already an existing feature.
+
+## Architecture
+
+The architecture for the frictionless airdrops follows the existing framework defined for handling all calls to the HederaTokenService system contract in the modularization services and is described in more detail in the Implementation section below.
+
+## Implementation
+
+### New Solidity Structures
+
+We will introduce the following new structures to support the new functionality:
+
+`PendingAirdrop` - A struct that represents a pending airdrop request.
+
+```solidity
+struct PendingAirdrop {
+    address sender;
+    address receiver;
+
+    address token;
+    int64 serial;
+}
+```
+
+`NftId` - A struct that represents the Nft serials to be rejected.
+
+```solidity
+struct NftId {
+    address nft;
+    int64[] serials;
```

Comment:
In follow up from note above this should be as follows in alignment with protobuf.
```suggestion
    int64 serial;
```